sys/include/net/wifi.h

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2023 Fabian Hüßler ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_wifi Wi-Fi
+ * @ingroup     net
+ * @brief       Provides Wi-Fi definitions
+ * @{
+ *
+ * @file
+ * @brief       Definitions for Wi-Fi
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ml-pa.com>
+ */
+
+#ifndef NET_WIFI_H
+#define NET_WIFI_H
+
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Wi-Fi region Europe
+ */
+#define WIFI_REGION_EUROPE              0
+
+/**
+ * @brief   Wi-Fi region North America
+ */
+#define WIFI_REGION_NORTH_AMERICA       1
+
+/**
+ * @brief   Wi-Fi region Asia
+ */
+#define WIFI_REGION_ASIA                2
+
+#if !defined(CONFIG_WIFI_REGION) || defined(DOXYGEN)
+/**
+ * @brief   Used to select regional Wi-Fi restrictions
+ */
+#define CONFIG_WIFI_REGION              WIFI_REGION_EUROPE
+#endif
+
+/**
+ * @brief   Current Wi-Fi region
+ */
+#define WIFI_REGION                     CONFIG_WIFI_REGION
+
+/**
+ * @brief   2.4 GHz channel 1
+ */
+#define WIFI_2_4_CH_1                   1
+
+/**
+ * @brief   2.4 GHz channel 2
+ */
+#define WIFI_2_4_CH_2                   2
+
+/**
+ * @brief   2.4 GHz channel 3
+ */
+#define WIFI_2_4_CH_3                   3
+
+/**
+ * @brief   2.4 GHz channel 4
+ */
+#define WIFI_2_4_CH_4                   4
+
+/**
+ * @brief   2.4 GHz channel 5
+ */
+#define WIFI_2_4_CH_5                   5
+
+/**
+ * @brief   2.4 GHz channel 6
+ */
+#define WIFI_2_4_CH_6                   6
+
+/**
+ * @brief   2.4 GHz channel 7
+ */
+#define WIFI_2_4_CH_7                   7
+
+/**
+ * @brief   2.4 GHz channel 8
+ */
+#define WIFI_2_4_CH_8                   8
+
+/**
+ * @brief   2.4 GHz channel 9
+ */
+#define WIFI_2_4_CH_9                   9
+
+/**
+ * @brief   2.4 GHz channel 10
+ */
+#define WIFI_2_4_CH_10                  10
+
+/**
+ * @brief   2.4 GHz channel 11
+ */
+#define WIFI_2_4_CH_11                  11
+
+/**
+ * @brief   2.4 GHz channel 12
+ */
+#define WIFI_2_4_CH_12                  12
+
+/**
+ * @brief   2.4 GHz channel 13
+ */
+#define WIFI_2_4_CH_13                  13
+
+/**
+ * @brief   2.4 GHz channel 14
+ */
+#define WIFI_2_4_CH_14                  14
```

Comment:
Please use whenever possible enumaration for constant definitions.
```suggestion
/**
 * @brief   2.4 GHz Channels
 */
enum {
    WIFI_2_4_CH_1 = 1       /**< 2.4 GHz channel 1 */
    WIFI_2_4_CH_2 = 2       /**< 2.4 GHz channel 2 */
    WIFI_2_4_CH_3 = 3       /**< 2.4 GHz channel 3 */
    WIFI_2_4_CH_4 = 4       /**< 2.4 GHz channel 4 */
    WIFI_2_4_CH_5 = 5       /**< 2.4 GHz channel 5 */
    WIFI_2_4_CH_6 = 6       /**< 2.4 GHz channel 6 */
    WIFI_2_4_CH_7 = 7       /**< 2.4 GHz channel 7 */
    WIFI_2_4_CH_8 = 8       /**< 2.4 GHz channel 8 */
    WIFI_2_4_CH_9 = 9       /**< 2.4 GHz channel 9 */
    WIFI_2_4_CH_10 = 10     /**< 2.4 GHz channel 10 */
    WIFI_2_4_CH_11 = 11     /**< 2.4 GHz channel 11 */
    WIFI_2_4_CH_12 = 12     /**< 2.4 GHz channel 12 */
    WIFI_2_4_CH_13 = 13     /**< 2.4 GHz channel 13 */
    WIFI_2_4_CH_14 = 14     /**< 2.4 GHz channel 14 */
};
```